### DISCUSSION TITLE
2022-02-07 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1049] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2114757)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2114762)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2114763)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2114765)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2114766)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2114770)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2124199)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2125993)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2127133)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2127166)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2128845)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2129379)
- [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/1049#discussioncomment-2130686)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### General Updates
- We're very close to closing the [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) milestone, we only have one issue left!
- We've merged in a few major projects, including:
	- API error handling standardization
	- Type configuration on the frontend

### Feedback Requested
- **Discussions in progress**:
	- https://github.com/centerofci/mathesar/discussions/1046
	- https://github.com/centerofci/mathesar/discussions/1035
	- https://github.com/centerofci/mathesar/discussions/1034
	- https://github.com/centerofci/mathesar/discussions/1007
- **Specs in review**:
	- _None at the moment_
- **PRs needing review**:
	- *Please see frontend and backend priorities sections.*

### Complete
- **Major wiki updates**:
	- We've been working on [Google Summer of Code project ideas](https://wiki.mathesar.org/en/community/mentoring/project-ideas)
	- The [Formulas](https://wiki.mathesar.org/en/product/specs/2022-01-views/07-formulas) page has been added to the Views product spec.
	- Front end coding standards update: https://github.com/centerofci/mathesar-wiki/pull/33
- **PRs merged**:
	- Full stack:
		- https://github.com/centerofci/mathesar/pull/1026
	- Backend:
		- https://github.com/centerofci/mathesar/pull/1016
		- https://github.com/centerofci/mathesar/pull/1017
		- https://github.com/centerofci/mathesar/pull/991
	- Frontend
		- https://github.com/centerofci/mathesar/pull/926
		- https://github.com/centerofci/mathesar/pull/1024
		- https://github.com/centerofci/mathesar/pull/963
		- https://github.com/centerofci/mathesar/pull/1005
		- https://github.com/centerofci/mathesar/pull/976
		- https://github.com/centerofci/mathesar/pull/961

**Comment by kgodey** [comment]
## Frontend priorities

1. Get Google Summer of Code ideas written up, this is top priority because of an external deadline.
2. Frontend work for #1041
3. Review and merge PRs ready for review. 
	1. https://github.com/centerofci/mathesar/pull/1048
4. Complete #775
5. Complete and merge basic E2E testing: https://github.com/centerofci/mathesar/issues/640
	1. PR: https://github.com/centerofci/mathesar/pull/1032
6. Make progress on issues from the [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) and [2022-02 improvements](https://github.com/centerofci/mathesar/milestone/43) milestones. Some ideas: 
	1. #500
	2. #742
	3. #970
	4. #928
	5. #929
	6. #936

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good.

I'll be prioritizing #775 and then #640. I expect those to take most/all of this week, especially since it's a 4-day week for me.

I don't have any GSoC project ideas, so I don't plan to spend any time with that.

I'm happy to do the front end work for #1041, but I'd like to give @pavish a chance to volunteer if he likes since I jumped on #862 and #1026 pretty quickly.

I plan to do the font end work on #1047 once it's ready.

Also, #619 is now pretty high on my priority list since @pavish and I have agreed on a simpler path towards closing that one, so I'd be keen to pick that up after resolving all of the stuff I mentioned above.

**Reply by pavish** [reply]
Sounds good to me.

@seancolsen I'll take up the frontend work for #1041

**Comment by kgodey** [comment]
## Backend priorities

The frontend is catching up fast to the backend, so we really need to get backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) completed and start on [Links between Tables](https://github.com/centerofci/mathesar/milestone/47) ASAP. We won't be able to do that this week, but the closer we get, the better.

1. Get Google Summer of Code ideas written up, this is top priority because of an external deadline.
2. **Filtering**:
	1. Get the following PRs merged:
		1. #1022
		2. #1041
	2. Create PR for new filters API and get that as close to merge as possible.
	3. Start working on individual data type filtering issues:
		1. https://github.com/centerofci/mathesar/issues/385
		2. https://github.com/centerofci/mathesar/issues/406
		3. https://github.com/centerofci/mathesar/issues/410
		4. https://github.com/centerofci/mathesar/issues/413
		5. https://github.com/centerofci/mathesar/issues/427
3. **Basic Data Type Support:**
	1. #430
	2. #420 
	3. #417
4. **Replace column name usage with ID:**
	1. #839
		1. #992
		2. #993
		3. #994
		4. #995
	2. #896
5. **Date and time spec**: #845
6. **Data Type API updates**:
	1. #960 
	2. #1036
	3. #1037
7. **Grouping**:
	1. #408 
	2. #411 
	3. #414
	4. #384
	5. #429
8. **Default values**: #648

That still leaves the following blocked issues, which we can get to once the all the issues under (3) are done:
1. #431
2. #432
3. #433
4. #418
5. #419

I didn't put #1008 in here but I'll be working on it if I have time this week.

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by silentninja** [reply]
I can work on issue #420 and then move on to text grouping issues.

**Comment by kgodey** [comment]
## Design priorities

1. Get Google Summer of Code ideas written up, this is top priority because of an external deadline.
2. Design specs for:
	1. Query builder - ticket will be created soon.
	2. #466
	3. #978 
	4. #1028
3. Continue exploratory work for Views in the design prototype.

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Comment by kgodey** [comment]
## Product priorities

1. Get Google Summer of Code ideas written up, this is top priority because of an external deadline.
2. Update Views design issues with requirements based on the  [new product spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views) and further exploration.

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-02-07 to 2022-02-11 | Sean |
| 2022-02-14 to 2022-02-18 | Dominykas |
| 2022-02-21 to 2022-02-25 | Ghislaine |
| 2022-02-28 to 2022-03-04 | Brent |
| 2022-03-07 to 2022-03-11 | Kriti |
| 2022-03-14 to 2022-03-18 | Pavish |
| 2022-03-21 to 2022-03-25 | Mukesh |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by dmos62** [comment]
> How did your most recent work go?

Nothing to report about my tasks.

I'm happy about the meta discussions around the feature-oriented v. architecture-oriented work. There's a sense of growth in the air.

> What are you planning to work on soon?

Get the function API and API namespace PRs merged. Make as much progress on filtering API as possible.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- Going well. I'm happy that we're all working on picking up the pace a bit.

> What are you planning to work on soon?

- See [my comment in "Frontend Priorities"](https://github.com/centerofci/mathesar/discussions/;1049#discussioncomment-2125987).

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

It went pretty good. Nothing major to report in particular.

> What are you planning to work on soon?

I'm planning to get the following done this week:

* Split frontend issues for initial data types into
  - Type switching with database options (includes sending type_config)
  - Implement display options
  - CellInput for the type
  - Filters (Common issue for all types)
  - Groups (Common issue for all types)
* Work on newly created issues for Boolean, Text and Number
* Number component improvements

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Reply by kgodey** [reply]
@pavish Will CellInput also cover styling issues (e.g. fixed width font for numbers?)

**Reply by pavish** [reply]
I initially planned to create a single PR for the CellInput component. I realized it would be better to handle it type by type. The first PR for Boolean might take a little longer than the rest, since it may include additional ground work.

**Reply by pavish** [reply]
@kgodey It will cover everything mentioned in the parent issue description. For [Number](https://github.com/centerofci/mathesar/issues/2580), monospace font is included in the issue description, so we will cover it.

It will introduce styling inconsistencies though, so any additional styling fixes can be taken up in the UI styling milestone, or as subsequent issues.

**Reply by kgodey** [reply]
Sounds good, thanks @pavish.

**Comment by mathemancer** [comment]
## Brent's check in

> How did your most recent work go?

Pretty slowly, since I was out for lunar New Year most of last week.  That said, I'm glad we synced a bit on how to increase velocity.  I'm still working on the interval type; it's taking longer than I thought, mostly due to trying to interpret some of the more esoteric sections of SQLAlchemy docs.

> What are you planning to work on soon?

I have the interval type working, I'm hooking it up to the API and testing.  After that, I'm on the grouping tasks.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

@dmos62 I'll hit you up this week to try to hook up a grouping function to the new functions API and hints framework.  I think if we can get that to work without too much fuss it'll save duplicated work.

**Reply by kgodey** [reply]
@mathemancer I think it would make sense to prioritize the Money types over grouping tasks - those are the last "data type support" issues to complete and will unblock all remaining grouping, filtering, and display options tasks.

**Reply by mathemancer** [reply]
Sure, I could do that.

**Reply by kgodey** [reply]
Actually, it seems like @silentninja is working on money types, so never mind.

**Comment by kgodey** [comment]
> How did your most recent work go?

It went well, I think @ghislaineguerin and I are finally figuring out a coherent version of how Views will fit together, there's fewer unknowns and I think most of them are known unknowns. The tickets and spec are out of date / half-finished / all over the place at the moment, I hope that by the end of the week, they will be more coherent.

I'm also glad we talked about increasing velocity, I'm looking forward to seeing us move faster.

> What are you planning to work on soon?

- Google Summer of Code ideas and coordination
- Hopefully finalizing Views product work and design tickets
- #1008

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

We all need to finish work on Google Summer of Code ideas soon, but that should be covered under individual priorities.

**Comment by ghislaineguerin** [comment]
### Ghislaine's Check-in

> How did your most recent work go?

Having more collaboration w product (@kgodey ) is helping to move forward and unblock design work related to views. I think it went well in terms of focus, and the scope is a lot more manageable than in previous weeks. 

> What are you planning to work on soon?

Continue exploratory work for Views in the design prototype.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by silentninja** [comment]
### Mukesh's Check-in

> How did your most recent work go?

It went well, finally got the error handling PR's merged and the integration test setup is going along quite well. 

> What are you planning to work on soon?

Looking to work on the Money type issue and text grouping issues.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None
